File: Java/String_2/countCode.java
/**
 * @author Luis Teixeira
 *
 * @date   29/03/2018
 *
 * @mail   filipe.teixeira.996@gmail.com
 */
package String_2;

import static org.junit.Assert.*;

import org.junit.Test;

/**

Return the number of times that the string "code" appears anywhere in the given string,
except we'll accept any letter for the 'd', so "cope" and "cooe" count.


countCode("aaacodebbb") --> 1
countCode("codexxcode") --> 2
countCode("cozexxcope") --> 2
 */
public class countCode {

	private static int i = 0;
	
	@Test
	public void Test1()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countCode(\"aaacodebbb\") --> 1\n");
		assertEquals("countCode(\"aaacodebbb\") --> 1", 1, countCode(\"aaacodebbb\"));
		i++;
	}

	@Test
	public void Test2()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countCode(\"codexxcode\") --> 2\n");
		assertEquals("countCode(\"codexxcode\") --> 2", 2, countCode(\"codexxcode\"));
		i++;
	}
	
	@Test
	public void Test3()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countCode(\"cozexxcope\") --> 2\n");
		assertEquals("countCode(\"cozexxcope\") --> 2", 2, countCode("cozexxcope"));
		i++;
	}
	public int countCode(String str) {
		  
	}

}

File: Python/String-2/count_code.py
import  unittest


'''

Return the number of times that the string "code" appears anywhere in the given string,
except we'll accept any letter for the 'd', so "cope" and "cooe" count.


count_code('aaacodebbb') -> 1
count_code('codexxcode') -> 2
count_code('cozexxcope') -> 2

'''

def count_code(str):
  
    substring1 = "code"
    substring2 = "cooe"
    substring3 = "cope"
    count      = 0
    
    count = str.count(substring1)
    count += str.count(substring2)
    count += str.count(substring3)
    
    return count
    


class Test_count_code(unittest.TestCase):

    def test_1(self):
        self.assertEqual(count_code('aaacodebbb'), 1)

    def test_2(self):
        self.assertEqual(count_code('codexxcode'), 2)

    def test_3(self):
        self.assertEqual(count_code('cozexxcope'), 2)


if __name__ == '__main__':
    unittest.main() 